src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationCodeFixProvider.cs

```diff
@@ -49,133 +49,151 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             // of the local.  This is necessary in some cases (for example, when the
             // type of the out-var-decl affects overload resolution or generic instantiation).
 
-            foreach (var diagnostic in diagnostics)
+            // Group by the ExpressionSyntax that the modified expression targets
+            var diagnosticGroups = diagnostics.GroupBy(d => d.AdditionalLocations[2]);
```

Comment:
the is a 'what' comment (i.e. it says what it is doing), but it's not a 'why' comment.  It doesn't tell me why this is appropriate or why it's better than not doing this.